docs/versioning.md

```diff
@@ -0,0 +1,239 @@
+mean that there should be no significant disruption to data-plane connectivity
+nor to policy enforcement, beyond the necessary disruption incurred by the
+restart of individual components:
+
+ * during the Antrea Controller restart, new policies will not be
+   processed. Because the Controller also runs the validation webhook for
+   [Antrea-native policies](antrea-network-policy.md), an attempt to create an
+   Antrea-native policy resource before the restart is complete may return an
+   error.
+ * during an Antrea Agent restart, the Node's data-plane will be impacted: new
+   connections to & from the Node will not be possible, and existing connections
+   may break.
+
+In particular, it should be possible to upgrade Antrea without compromising
+enforcement of existing network policies for both new and existing Pods.
+
+In order to achieve this, the different Antrea components need to support
+version skew.
+
+ * **Antrea Controller**: must be upgraded first
+ * **Antrea Agent**: must not be newer than the **Antrea Controller**, and may
+   be up to 4 minor versions older
+ * **Antctl**: must not be newer than the **Antrea Controller**, and may be up
+   to 4 minor versions older
+
+The supported version skew means that we only recommend Antrea upgrades to a new
+release up to 4 minor versions newer. For example, a cluster using 0.10 can be
+upgraded to one of 0.11, 0.12, 0.13 or 0.14, but we discourage direct upgrades
+to 0.15 and beyond. With the current release cadence, this provides a 6-month
+window of compatibility. If we reduce our release cadence in the future, we may
+revisit this policy as well.
+
+When directly applying a newer Antrea YAML manifest, as provided for each
+[release](https://github.com/vmware-tanzu/antrea/releases), there is no
+guarantee that the Antrea Controller will be upgraded first. In practice, the
+Controller would be upgraded simultaneously with the first Agent(s) to be
+upgraded by the rolling update of the Agent DaemonSet. This may create some
+transient issues and compromise the "graceful" upgrade. For upgrade scenarios,
+we therefore recommend that you "split-up" the manifest to ensure that the
+Controller is upgraded first.
+
+## Supported K8s versions
+
+Each Antrea minor release should support [maintained K8s
+releases](https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions)
+at the time of release (3 up to K8s 1.19, 4 after that). For example, at the
+time that Antrea 0.10 was released, the latest K8s version was 1.19; as a result
+we guarantee that 0.10 supports at least 1.19, 1.18 and 1.17 (in practice it
+also supports K8s 1.16).
+
+In addition, we strive to support the K8s versions used by default in
+cloud-managed K8s services ([EKS], [AKS] and [GKE] regular channel).
+
+## Deprecation policies
+
+### Prometheus metrics deprecation policy
+
+Antrea follows a similar policy as
+[Kubernetes](https://cloud.google.com/kubernetes-engine/docs/release-notes) for
```

Comment:
I could not find any Prometheus deprecation policy for K8s in the link https://cloud.google.com/kubernetes-engine/docs/release-notes. Wondering if that is the intended link.

Reply:
Thanks. Looks like I got my links confused, will fix